components/director/internal/tenantfetchersvc/subscriber.go

```diff
@@ -0,0 +1,124 @@
+		ConsumerSubaccountIDsLabelKey: consumerSubaccountIDsLabelKey,
+	}
+}
+
+// Subscribe subscribes tenant to runtime. If the tenant does not exist it will be created
+func (s *subscriber) Subscribe(ctx context.Context, tenantSubscriptionRequest *TenantSubscriptionRequest, region string) error {
+	if err := s.provisioner.ProvisionTenants(ctx, tenantSubscriptionRequest, region); err != nil {
+		return err
+	}
+
+	return s.applyRuntimesSubscriptionChange(ctx, tenantSubscriptionRequest.SubscriptionConsumerID, tenantSubscriptionRequest.SubaccountTenantID, region, addElement)
+}
+
+// Unsubscribe unsubscribes tenant from runtime.
+func (s *subscriber) Unsubscribe(ctx context.Context, tenantSubscriptionRequest *TenantSubscriptionRequest, region string) error {
+	return s.applyRuntimesSubscriptionChange(ctx, tenantSubscriptionRequest.SubscriptionConsumerID, tenantSubscriptionRequest.SubaccountTenantID, region, removeElement)
+}
+
+func (s *subscriber) applyRuntimesSubscriptionChange(ctx context.Context, subscriptionConsumerID, subaccountTenantID, region string, mutateSliceFunc sliceMutationFunc) error {
+	filters := []*labelfilter.LabelFilter{
+		labelfilter.NewForKeyWithQuery(s.SubscriptionConsumerLabelKey, fmt.Sprintf("\"%s\"", subscriptionConsumerID)),
+		labelfilter.NewForKeyWithQuery(s.RegionLabelKey, fmt.Sprintf("\"%s\"", region)),
+	}
+
+	runtimes, err := s.ListByFiltersGlobal(ctx, filters)
+	if err != nil {
+		if apperrors.IsNotFoundError(err) {
+			return nil
+		}
+
+		return errors.Wrap(err, fmt.Sprintf("Failed to get runtimes for labels %s: %s and %s: %s", s.RegionLabelKey, region, s.SubscriptionConsumerLabelKey, subscriptionConsumerID))
+	}
+
+	for _, runtime := range runtimes {
+		ctx = tenant.SaveToContext(ctx, runtime.Tenant, "")
+
+		labelOldValue := make([]string, 0)
+		label, err := s.GetLabel(ctx, runtime.ID, s.ConsumerSubaccountIDsLabelKey)
+		if err != nil {
+			if !apperrors.IsNotFoundError(err) {
+				return errors.Wrap(err, fmt.Sprintf("Failed to get label for runtime with id: %s and key: %s", runtime.ID, s.ConsumerSubaccountIDsLabelKey))
+			}
+			// if the error is not found, do nothing and continue
+		} else {
+			if labelOldValue, err = labelutils.ValueToStringsSlice(label.Value); err != nil {
+				return errors.Wrap(err, fmt.Sprintf("Failed to parse label values for label with id: %s", label.ID))
+			}
+		}
+
+		var labelNewValue = mutateSliceFunc(labelOldValue, subaccountTenantID)
+
+		if err := s.SetLabel(ctx, &model.LabelInput{
+			Key:        s.ConsumerSubaccountIDsLabelKey,
+			Value:      labelNewValue,
+			ObjectType: model.RuntimeLabelableObject,
+			ObjectID:   runtime.ID,
+		}); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("Failed to set label for runtime with id: %s", runtime.ID))
+		}
+	}
+
+	return nil
+}
+
+func addElement(slice []string, elem string) []string {
+	return append(slice, elem)
+}
+
+func removeElement(slice []string, elem string) []string {
+	result := make([]string, 0)
```

Comment:
```suggestion
result := make([]string, len(slice))
```

This way **result** won't be resized multiple times during the for loop. In fact, it won't ever resize.